MetaMask/metamask-extension
# Convert action-queue to TypeScript

## Explanation
Converts the action-queue to typescript, which was needed to effectively type the action.js file.

## Manual Testing Steps
1. Pull branch
2. `yarn`
3. `yarn start`
4. Perform actions that hit the background such as settings toggles, etc
5. Ensure no change in behavior.

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
